pandas/core/computation/eval.py

```diff
@@ -384,7 +384,10 @@ def eval(
             try:
                 with warnings.catch_warnings(record=True):
                     # TODO: Filter the warnings we actually care about here.
-                    target[assigner] = ret
+                    if hasattr(target, "loc"):
```

Comment:
This is not ideal. What could target be here? Dict, series, DataFrame, something else?